discovery-provider/src/queries/get_challenges.py

```diff
@@ -0,0 +1,127 @@
+from functools import reduce
+from collections import defaultdict
+from typing import List, DefaultDict, TypedDict, Tuple
+from sqlalchemy import and_
+from src.models import UserChallenge, Challenge, ChallengeType, ChallengeDisbursement
+
+
+class ChallengeResponse(TypedDict):
+    challenge_id: str
+    user_id: int
+    specifier: str
+    is_complete: bool
+    is_active: bool # need this
+    is_disbursed: bool # need this
+    current_step_count: int
+    max_steps: int
+    challenge_type: str
+
+def rollup_aggregates(user_challenges: List[UserChallenge], challenge: Challenge) -> ChallengeResponse:
+    num_complete = reduce(lambda acc, cur: acc + 1 if cur.is_complete else acc, user_challenges, 0)
+    is_complete = num_complete >= challenge.step_count
```

Comment:
is >= necessary? or just for completeness?

Reply:
Hopefully not, but playing it safe here in case there is a race or something and we create more partial completions than we allow.